components/dashboard/src/app/AppBlockingFlows.tsx

```diff
@@ -17,13 +17,20 @@ const DedicatedSetup = lazy(() => import(/* webpackPrefetch: true */ "../dedicat
 
 // This component handles any flows that should come after we've loaded the user/orgs, but before we render the normal app chrome.
 // Since this runs before the app is rendered, we should avoid adding any lengthy async calls that would delay the app from loading.
-export const AppBlockingFlows: FC = ({ children }) => {
+export const AppBlockingFlows = ({ children, onReady }: { children: ReactNode; onReady?: () => void }) => {
     const history = useHistory();
     const user = useCurrentUser();
     const org = useCurrentOrg();
     const showDedicatedSetup = useShowDedicatedSetup();
     const showUserOnboarding = useShowUserOnboarding();
 
+    useEffect(() => {
+        // once user is loaded, page is not blocking
+        if (onReady && user) {
+            onReady();
+        }
+    }, [user, onReady]);
```

Comment:
Suspend don't have onReady event, since `once user is loaded, page is not blocking`, we consider it as ready